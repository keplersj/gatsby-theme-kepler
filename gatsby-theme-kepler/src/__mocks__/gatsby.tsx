import * as React from "react";
const gatsby = jest.requireActual("gatsby");

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn(
    ({
      // these props are invalid for an `a` tag
      /* eslint-disable @typescript-eslint/no-unused-vars */
      /* eslint-disable no-unused-vars */
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      /* eslint-enable @typescript-eslint/no-unused-vars */
      /* eslint-enable no-unused-vars */
      to,
      ...rest
      // eslint-disable-next-line jsx-a11y/anchor-has-content
    }): React.ReactElement => <a {...rest} href={to} />
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn()
};
