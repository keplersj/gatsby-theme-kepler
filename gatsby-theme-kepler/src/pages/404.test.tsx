import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import Page from "./404";
import {
  KeplerBannerBackgroundData,
  KeplerBaseLayoutData
} from "../__mockData__";
import { HelmetProvider } from "react-helmet-async";

// HelmetProvider.canUseDom = false;

beforeEach((): void => {
  (useStaticQuery as jest.Mock).mockImplementation((): object => ({
    ...KeplerBannerBackgroundData,
    ...KeplerBaseLayoutData
  }));
});

describe("404 Page", (): void => {
  it("renders correctly", (): void => {
    const context: { helmet?: object } = {};
    const tree = renderer
      .create(
        <HelmetProvider context={context}>
          <Page />
        </HelmetProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(context.helmet).toMatchSnapshot();
  });
});
