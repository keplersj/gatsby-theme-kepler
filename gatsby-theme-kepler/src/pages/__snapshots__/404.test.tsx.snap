// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page renders correctly 1`] = `
.emotion-2 {
  position: relative;
}

.emotion-2::before,
.emotion-2::after {
  content: "";
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  -webkit-transition: opacity 0.5s ease 0.25s;
  transition: opacity 0.5s ease 0.25s;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.emotion-2::before {
  z-index: -101;
  opacity: 1;
  background-image: url("data+image/svg+xml");
}

@media screen and (prefers-color-scheme:dark) {
  .emotion-2::before {
    background-image: url("data+image/svg+xml");
  }
}

.emotion-0 {
  opacity: 1;
  background-image: url("/static/banner_example/3/banner.jpg");
}

@media screen and (prefers-color-scheme:dark) {
  .emotion-0 {
    background-image: url("/static/banner_dark_example/3/banner_dark.jpg");
  }
}

.emotion-1 {
  color: #505050;
  font-size: 1.5em;
  font-weight: 500;
  line-height: 1.5;
  height: 100vh;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
}

@media (prefers-color-scheme:dark) {
  .emotion-1 {
    color: white;
  }
}

<div
  className="emotion-2"
  data-testid="BackgroundImageContainerElement"
>
  <noscript
    className="emotion-0"
  />
  <div
    className="emotion-1"
  >
    <div>
      <h1>
        404!
      </h1>
      <span>
        Maybe, go back to the 
        <a
          href="/"
        >
          home page
        </a>
        ?
      </span>
    </div>
  </div>
</div>
`;

exports[`404 Page renders correctly 2`] = `
<html
  lang="en"
>
  <head>
    <meta
      content=""
      data-rh={true}
      name="description"
    />
    <meta
      content="Kepler Sticka-Jones"
      data-rh={true}
      property="og:site_name"
    />
    <meta
      content="404!"
      data-rh={true}
      property="og:title"
    />
    <meta
      content=""
      data-rh={true}
      property="og:description"
    />
    <meta
      content="summary"
      data-rh={true}
      name="twitter:card"
    />
    <meta
      content="@test"
      data-rh={true}
      name="twitter:site"
    />
    <meta
      content="@test"
      data-rh={true}
      name="twitter:creator"
    />
    <title
      data-rh={true}
    >
      404! | Kepler Sticka-Jones
    </title>
  </head>
  <body />
</html>
`;
